// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package privacy_sandbox.server_common.gvisor;

import "src/roma/interface/function_binding_io.proto";

option go_package = "github.com/privacysandbox/data-plane-shared/apis/roma/binary/callback";

message Callback {
  // Proto containing input/output args for native function.
  google.scp.roma.proto.FunctionBindingIoProto io_proto = 1;

  // User set per-callback request ID.
  optional string id = 2;

  // Name of native function to be invoked in host process
  string function_name = 3;
}
